index.js

```
@@ -0,0 +1,257 @@
+}
+// GET South America
+const sAmericaTrips = () => {
+  let url = (URL + '/continent?query=South%20America')
+  apiGET(url)
+}
+
+
+// SELECT/SEE TRIP INFO
+const getTrip = (id) => {
+  let url = URL + '/' + id
+  const chosenTrip = $('#chosen-trip');
+  chosenTrip.empty()
+  const reservedTrip = $('#trip-id');
+  reservedTrip.empty()
+
+  axios.get(url)
+  .then((response) => {
+    console.log(response)
+    chosenTrip.append(`<h1>🏜 ${response.data.name} 🏜</h1><p>Continent: ${response.data.continent}</p><p>Weeks: ${response.data.weeks}</p><p>Cost: $${response.data.cost}</p><p>About: ${response.data.about}</p>`);
+
+    reservedTrip.append(`<input type="hidden" name="trip_id" value="${response.data.id}" />`)
+
+    reportStatus('Trip Loaded!')
+    $('#additionals').removeClass(['hidden'])
+  })
+  .catch((error) => {
+    console.log(error);
+    reportStatus(`Error: ${error.message}`)
+  })
+}
+
+// RESERVE TRIP HELPERS
+const FORM_FIELDS = ['name', 'email', 'trip_id'];
+const inputField = name => $(`#trip-form input[name="${name}"]`);
+
+const readFormData = () => {
+  const getInput = name => {
+    const input = inputField(name).val();
+    return input ? input : undefined;
+  };
+
+  const formData = {};
+  FORM_FIELDS.forEach((field) => {
+    formData[field] = getInput(field);
+  });
+
+  return formData;
+};
+
+const clearForm = (array) => {
+  array.forEach((field) => {
+    inputField(field).val('');
+  });
+}
+
+
+// RESERVE TRIP
+const reserveTrip = (event) => {
+  event.preventDefault();
+  console.log(event);
+  const tripData = readFormData()
+  console.log(tripData);
+  let url = URL + '/' + tripData.trip_id + '/reservations'
+
+  axios.post(url, tripData)
+  .then((response) => {
+    console.log(response);
+    reportStatus(`Successfully reserved trip for ${response.data.name}!`);
+    clearForm(FORM_FIELDS);
+  })
+  .catch((error) => {
+    console.log(error.response);
+    if (error.response.data && error.response.data.errors) {
```

Hmmm... here you should also report any validation errors from the API.